tests/genai/optimize/test_optimize.py

```diff
@@ -253,6 +253,39 @@ def optimize(self, eval_fn, dataset, target_prompts):
     assert len(result.optimized_prompts) == 1
 
 
+def test_optimize_prompts_warns_on_unused_prompt(
+    sample_translation_prompt, sample_summarization_prompt, sample_dataset, capsys
+):
+    mock_optimizer = MockPromptAdapter()
+
+    # Create predict_fn that only uses translation prompt, not summarization prompt
+    def predict_fn_single_prompt(input_text, language):
+        prompt = mlflow.genai.load_prompt("prompts:/test_translation_prompt/1")
+        prompt.template
+        return sample_predict_fn(input_text=input_text, language=language)
+
+    result = optimize_prompts(
+        predict_fn=predict_fn_single_prompt,
+        train_data=sample_dataset,
+        prompt_uris=[
+            f"prompts:/{sample_translation_prompt.name}/{sample_translation_prompt.version}",
+            f"prompts:/{sample_summarization_prompt.name}/{sample_summarization_prompt.version}",
+        ],
+        optimizer=mock_optimizer,
+        scorers=[equivalence],
+    )
+
+    assert len(result.optimized_prompts) == 2
+
+    captured = capsys.readouterr()
+    assert "prompts were not used during evaluation" in captured.err, (
+        "Expected warning about unused prompts not found in stderr"
+    )
+    assert "test_summarization_prompt" in captured.err, (
+        "Expected 'test_summarization_prompt' in warning message"
+    )
```

Comment:
```suggestion
    assert "test_summarization_prompt" in captured.err
```

let's remove the messages. They just repeat what the assert statement says.